docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,194 @@ weight: 700
-  - [Groups / Teams](#groups--teams)
-  - [Login](#login)
-  - [PKCE](#pkce)
-  - [Configure refresh token](#configure-refresh-token)
-  - [Configure automatic login](#configure-automatic-login)
-  - [Set up OAuth2 with Auth0](#set-up-oauth2-with-auth0)
-  - [Set up OAuth2 with Bitbucket](#set-up-oauth2-with-bitbucket)
-  - [Set up OAuth2 with Centrify](#set-up-oauth2-with-centrify)
-  - [Set up OAuth2 with OneLogin](#set-up-oauth2-with-onelogin)
-  - [Role Mapping](#role-mapping)
-    - [JMESPath examples](#jmespath-examples)
-      - [Map user organization role](#map-user-organization-role)
-      - [Map server administrator privileges](#map-server-administrator-privileges)
-  - [Team synchronization](#team-synchronization)
-  - [Skip organization role sync](#skip-organization-role-sync)
-
-This callback URL must match the full HTTP address that you use in your browser to access Grafana, but with the suffixed path of `/login/generic_oauth`.
-
-You may have to set the `root_url` option of `[server]` for the callback URL to be
-correct. For example in case you are serving Grafana behind a proxy.
-
-Example config:
+{{< docs/shared "auth/intro.md" >}}
+
+Grafana has specialised OAuth2 integrations for the following auth providers:
+
+- [Azure AD OAuth]({{< relref "../azuread" >}})
+- [GitHub OAuth]({{< relref "../github" >}})
+- [GitLab OAuth]({{< relref "../gitlab" >}})
+- [Google OAuth]({{< relref "../google" >}})
+- [Grafana Com OAuth]({{< relref "../grafana-com" >}})
+- [Okta OAuth]({{< relref "../okta" >}})
+
+If your OAuth2 provider is not among them, you can use generic OAuth2 authentication to integrate it with Grafana.
+
+## Before you begin
+
+- Ensure that you have access to the [Grafana configuration file]({{< relref "docs/grafana/latest/setup-grafana/configure-grafana#configuration-file-location" >}}).
+- Learn how to create an OAuth2 application with your chosen OAuth2 provider. Consult the documentation of your OAuth2 provider for more information.
+- (Optional) Learn about how to set up refresh tokens with your OAuth2 provider. Consult the documentation of your OAuth2 provider for more information.
+
+## Steps
+
+To integrate your OAuth2 provider with Grafana using our generic OAuth2 authentication, follow these steps:
+
+1. Create an OAuth2 application in your chosen OAuth2 provider.
+1. Set the callback URL for your OAuth2 app to `http://<my_grafana_server_name_or_ip>:<grafana_server_port>/login/generic_oauth`.
+
+   This callback URL must match the full HTTP address that you use in your browser to access Grafana, but with the suffixed path of `/login/generic_oauth`.
+   You may need to set the `root_url` option of `[server]` for the callback URL to be correct. For example, in case you are serving Grafana behind a proxy.
+
+1. Update `[auth.generic_oauth]` section of Grafana configuration file:
+   1. Update `client_id` and `client_secret` fields to match client ID and client secret from your OAuth2 app.
```

Comment:
And then here, let's add a table that lists all of the fields.

| Field name | Description  |
|---|---|
| `client_id`, `client_secret  | These field values must match ...   | 
| `auth_url` | Set this field to the token endpoint ...  |

etc, etc.